src/test/kotlin/com/exactpro/th2/common/schema/TestJsonConfiguration.kt

```diff
@@ -49,113 +46,246 @@ class TestJsonConfiguration {
+                                    FieldFilterConfiguration(
+                                        "session_alias",
+                                        "codec-fix",
+                                        FilterOperation.EQUAL
+                                    )
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+
+        )
     }
 
     @Test
     fun `test backward compatibility for cradle confidential json configuration`() {
-        OBJECT_MAPPER.readValue(
+        testDeserialize(
             "{\"dataCenter\":\"datacenter1\",\"host\":\"localhost\",\"port\":\"9042\",\"keyspace\":\"schema_dev\",\"username\":\"th2\",\"password\":\"CASSANDRA_PASS\"}",
-            CradleConfidentialConfiguration::class.java
+            CradleConfidentialConfiguration("datacenter1", "localhost", "schema_dev", 9042, "th2", "CASSANDRA_PASS")
         )
     }
+
     @Test
     fun `test cradle non confidential json configuration`() {
-        testJson(CradleNonConfidentialConfiguration())
+        testSerializeAndDeserialize(CradleNonConfidentialConfiguration())
     }
 
     @Test
     fun `test prometheus confidential json configuration`() {
-        testJson(PrometheusConfiguration())
-        OBJECT_MAPPER.readValue(
-            "{\"enabled\":\"true\"}",
-            PrometheusConfiguration::class.java)
+        testSerializeAndDeserialize(PrometheusConfiguration())
+
+        testDeserialize(
+            "{\"enabled\":\"false\"}",
+            PrometheusConfiguration(enabled = false)
+        )
+    }
+
+    @Test
+    fun `test new filter format json configuration`() {
+        testDeserialize(
+            "{\"queues\":{\"pin0\":{\"attributes\":[\"publish\",\"event\"],\"exchange\":\"demo_exchange\",\"filters\":[{\"message\":[{\"fieldName\":\"field0\",\"expectedValue\":\"value0\",\"operation\":\"EQUAL\"}]}],\"name\":\"key[schema-dev:act-dev-entry-point:estore-pin]\",\"queue\":\"\"}}}",
+            MessageRouterConfiguration(
+                mapOf(
+                    "pin0" to QueueConfiguration(
+                        "key[schema-dev:act-dev-entry-point:estore-pin]",
+                        "",
+                        "demo_exchange",
+                        listOf("publish", "event"),
+                        listOf(
+                            MqRouterFilterConfiguration(
+                                emptyList(),
+                                listOf(FieldFilterConfiguration("field0", "value0", FilterOperation.EQUAL))
+                            )
+                        )
+                    )
+                )
+            )
+        )
     }
 
-    private fun testJson(configuration: Any) {
-        OBJECT_MAPPER.writeValueAsString(configuration).also {
-            assertEquals(configuration, OBJECT_MAPPER.readValue(it, configuration::class.java))
+    private fun testSerializeAndDeserialize(configuration: Any) {
```

Comment:
I still think that this kind of test does not check anything. If our bean can be serialized and after that deserialized does not mean that the JSON that is produced by infra-operator will be correctly deserialized.
To make sure that they will need to use them as a source for deserialization. It is still only my opinion and if you disagree we can leave it as it is for now